packages/webapp/pages/_app.tsx

```diff
@@ -167,29 +152,7 @@ function InternalApp({ Component, pageProps, router }: AppProps): ReactElement {
     };
   }, [modal, closeModal, router.events]);
 
-  useQuery({
-    queryKey: generateQueryKey(RequestKey.CheckCoresRole, user),
-    queryFn: async () => {
-      const result = await gqlClient.request<{
-        checkCoresRole: Pick<LoggedUser, 'coresRole'>;
-      }>(CHECK_CORES_ROLE_QUERY);
-
-      if (result.checkCoresRole.coresRole !== user.coresRole) {
-        await updateUser({
-          ...user,
-          coresRole: result.checkCoresRole.coresRole,
-        });
-      }
-
-      return result.checkCoresRole;
-    },
-    staleTime: Infinity,
-    gcTime: Infinity,
-    enabled:
-      !!user &&
-      isActionsFetched &&
-      !checkHasCompleted(ActionType.CheckedCoresRole),
-  });
+  useCheckCoresRole();
```

Comment:
Group it with the other hooks (before useEffect)?

Reply:
done